### PR TITLE
terarium dataservice migration changes

### PR DESCRIPTION
**Pull request by dvince2**
This PR handles the changes needed to move to the new data service endpoints. This branch is untested currently, and contains the following changes:

- Changing endpoints and parameters to use `kebab-case`
- JSON keys are serialized / deserialized from camelCase rather than snake_case by default. Support for a "X-Enable-Snake-Case" header has been integrated which when provided will serialize / deserialize JSON keys as snake_case
- Our backend uses OAuth2 and many endpoints require a user object for ReBAC. For simplicity we have created a shared service-user that will be injected if the correct basic auth credential is added to each request. I've introduced TDS_USER and TDS_PASSWORD env vars that will be injected in a basic auth header in each request. Please contact our team and we will provide the credential for our staging env. Please do not commit the credential to the repo unencrypted.

### CODE REVIEW

File: reporting/tests/seed.py [hunk 1]
```diff
@@ -25,17 +26,16 @@ def create_project():
     project = {
         "name": "Integration Test Suite Project",
         "description": f"Test generated at {ts}",
-        "assets": [],
-        "active": True,
```
Comment (Sorrento110) [inline reporting/tests/seed.py:29]: Did we remove the active tag on projects in the HMI server? Or will it just default?

File: reporting/tests/seed.py [hunk 2]
```diff
@@ -87,21 +84,20 @@ def add_code(scenario, project_id, file_path_override=None):
...
-                headers={"Content-Type": "application/json"},
             )
 
             if code_response.status_code >= 300:
                 raise Exception(
                     f"Failed to POST code ({code_response.status_code}): {scenario}"
                 )
             else:
-                add_asset(_id, "code", project_id)
+                add_asset(code_response.json()['id'], "code", project_id)
```
Comment (Sorrento110) [inline reporting/tests/seed.py:97]: Unfortunately I think that if we are getting a dynamic uuid back from the HMI server here the rest of the tests will break. I believe they are still dependent on the ids being specified by the directory structure in the tests. So when we generate the id above in the old code with `_id = f"{project_id}-{scenario.replace(' ','-')}"` this is also the expected id structure further down the line for the rest of the tests.  Any id changes that don't follow this expectation may break the tests, but I would have to run this new code to be sure.

**Review by Sorrento110 — COMMENTED** [review]
Looks good overall, but I think some of the reporting changes to seeding may break the rest of the reporting flow.

File: reporting/tests/seed.py (hunk 1 above)
Comment (dgauldie) [reply reporting/tests/seed.py:29]: yeah we removed them. We now use `createdOn`, `updatedOn`, and `deletedOn`

File: reporting/tests/seed.py (hunk 2 above)
Comment (dgauldie) [reply reporting/tests/seed.py:97]: I think that because you mock TDS calls, tests will still pass

**Comment by dgauldie** [comment]
@Sorrento110 and @fivegrant , how do we test the report generation on this branch? can one of you see if we've broken things before we merge this?

**Comment by fivegrant** [comment]
@dgauldie take a look at this https://github.com/DARPA-ASKEM/knowledge-middleware/tree/main/reporting

**Comment by fivegrant** [comment]
@dgauldie just tried to run the above but the testing depends on standing up and seeding TDS in the docker compose just like simulation-integration. Report generation won't work unless [HMI Server replaces TDS in the docker compose](https://github.com/DARPA-ASKEM/knowledge-middleware/blob/090e0701e4330b0c8df7ef4aa793c01465fe4ec3/reporting/docker-compose.yaml#L6-L34)

**Comment by fivegrant** [comment]
@dgauldie i just pushed a fix so the tests can run. reporting was broken because stuff was moved around at project root and data service was still being referenced in a few places

**Review by YohannParis — APPROVED** [review]

**Pull request merged**